tools/oci/go_image.bzl

```diff
@@ -0,0 +1,33 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_binary")
+load("@rules_pkg//:pkg.bzl", "pkg_tar")
+load("@rules_oci//oci:defs.bzl", "oci_image")
+
+def go_image(name, base = "@distroless_base", srcs = [], embed = [], tags = [], visibility = None, pure = "on", arch = "amd64", os = "linux"):
+    go_binary(
+        name = "{}_binary".format(name),
+        srcs = srcs,
+        embed = embed,
+        goarch = arch,
+        goos = os,
+        pure = pure,
+        tags = tags,
+        visibility = visibility,
+    )
+
+    pkg_tar(
```

Comment:
This seems like a lot of manual work. Compared to rules_docker’s approach it likely has the downside that runfiles also don’t work out of the box. (Though we fortunately don’t depend on that.)

Is rules_oci planning on streamlining that?

Reply:
rules_docker's approach has its downsides as well. it's a huge maintenance burden to us, to take language rules as dependencies and successfully maintain them. rules_docker failed to do so. 

we have two approaches here 

1- design your binaries to not rely on runfiles. statically built languages are less likely to be already dependent on runfiles. 
2- have a rule that outputs bazel-like runfiles layout in a tar file. we have experimented with this in [here](https://github.com/bazel-contrib/rules_oci/blob/122d1b181e3b6d71fed75eb7cf7b61f788b38b82/language/runfiles.bzl). it's a generic helper rule that is capable of transitioning to the target platform and laying out the runfiles tree as a tar layer.

Reply:
generic `runfiles` helper and `pkg_tar` is the superior choice and we'll be spending our energy fixing bazel and rules_tar to be able to yield a runfiles tree.

Reply:
@EdSchouten I can come back and remove this once we support transitions upstream.

Reply:
To be clear, pkg_tar does support `include_runfiles` https://github.com/bazelbuild/rules_pkg/pull/398